ksqldb-udf/src/main/java/io/confluent/ksql/schema/ksql/types/SqlLambda.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.schema.ksql.types;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.errorprone.annotations.Immutable;
+import io.confluent.ksql.schema.utils.FormatOptions;
+import java.util.Objects;
+
+@Immutable
+public final class SqlLambda extends SqlType {
+
+  private final SqlType inputType;
```

Comment:
Just for my own understanding: how to represent `(X, Y) -> X + Y` with a single input type?

Reply:
See comment above.